app/notifications/validators.py

```diff
@@ -99,17 +105,88 @@ def check_service_over_daily_message_limit(key_type: ApiKeyType, service: Servic
     counter_name="rate_limit.live_service_daily_sms",
     exception=LiveServiceTooManySMSRequestsError,
 )
-def check_service_over_daily_sms_limit_and_warn(key_type: ApiKeyType, service: Service):
-    if current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"] and key_type != KEY_TYPE_TEST and current_app.config["REDIS_ENABLED"]:
-        fragments_sent = fetch_daily_sms_fragment_count(service.id)
-        warn_about_daily_sms_limit(service, fragments_sent)
+def check_sms_daily_limit(service: Service, key_type: ApiKeyType = KEY_TYPE_NORMAL):
+    if not current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"]:
+        return
+    if key_type == KEY_TYPE_TEST:
+        return
+    if not current_app.config["REDIS_ENABLED"]:
+        return
+    messages_sent = fetch_daily_sms_fragment_count(service.id)
+    over_sms_daily_limit = messages_sent >= service.sms_daily_limit
+
+    # Send a warning when reaching the daily message limit
+    if not over_sms_daily_limit:
+        return
+
+    current_app.logger.info(
+        f"service {service.id} has been rate limited for daily sms use sent {int(messages_sent)} limit {service.sms_daily_limit}"
+    )
+    if service.restricted:
+        raise TrialServiceTooManySMSRequestsError(service.sms_daily_limit)
+    else:
+        raise LiveServiceTooManySMSRequestsError(service.sms_daily_limit)
+
+
+def send_warning_sms_limit_emails_if_needed(service: Service, key_type: ApiKeyType = KEY_TYPE_NORMAL):
+    if not current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"]:
+        return
+    if key_type == KEY_TYPE_TEST:
+        return
+    if not current_app.config["REDIS_ENABLED"]:
+        return
+    messages_sent = fetch_daily_sms_fragment_count(service.id)
```

Comment:
I think we should add one to represent the current notification being sent.  That way, the user will get the warning when they send their 8th email when their limit is 10 rather than their 9th.  Thoughts?

```suggestion
    messages_sent = fetch_daily_sms_fragment_count(service.id) + 1
```

Reply:
Nevermind, you already handle this by considering the `requested_sms`!